internal/requests/username_password_request.go

```diff
@@ -5,55 +5,57 @@ package requests
 
 import (
 	"errors"
+	"log"
 
 	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/msalbase"
 )
 
 // UsernamePasswordRequest stuff
 type UsernamePasswordRequest struct {
 	webRequestManager WebRequestManager
-	authParameters    *msalbase.AuthParametersInternal
+	authParameters    msalbase.AuthParametersInternal
 }
 
 // CreateUsernamePasswordRequest stuff
-func CreateUsernamePasswordRequest(
-	webRequestManager WebRequestManager,
-	authParameters *msalbase.AuthParametersInternal) *UsernamePasswordRequest {
+func CreateUsernamePasswordRequest(webRequestManager WebRequestManager, authParameters msalbase.AuthParametersInternal) *UsernamePasswordRequest {
 	req := &UsernamePasswordRequest{webRequestManager, authParameters}
 	return req
 }
 
-// Execute stuff
-func (req *UsernamePasswordRequest) Execute() (*msalbase.TokenResponse, error) {
-
+func (req *UsernamePasswordRequest) Execute() (msalbase.TokenResponse, error) {
 	resolutionManager := CreateAuthorityEndpointResolutionManager(req.webRequestManager)
 	endpoints, err := resolutionManager.ResolveEndpoints(req.authParameters.AuthorityInfo, "")
 	if err != nil {
-		return nil, err
+		return msalbase.TokenResponse{}, err
 	}
 
 	req.authParameters.Endpoints = endpoints
 
 	userRealm, err := req.webRequestManager.GetUserRealm(req.authParameters)
 	if err != nil {
-		return nil, err
+		return msalbase.TokenResponse{}, err
 	}
 
 	switch accountType := userRealm.GetAccountType(); accountType {
 	case msalbase.Federated:
-		if mexDoc, err := req.webRequestManager.GetMex(userRealm.FederationMetadataURL); err == nil {
-			wsTrustEndpoint := mexDoc.UsernamePasswordEndpoint
-			if wsTrustResponse, err := req.webRequestManager.GetWsTrustResponse(req.authParameters, userRealm.CloudAudienceURN, &wsTrustEndpoint); err == nil {
-				if samlGrant, err := wsTrustResponse.GetSAMLAssertion(&wsTrustEndpoint); err == nil {
-					return req.webRequestManager.GetAccessTokenFromSamlGrant(req.authParameters, samlGrant)
-				}
-			}
+		mexDoc, err := req.webRequestManager.GetMex(userRealm.FederationMetadataURL)
+		if err != nil {
+			// TODO(someone): check for ui interaction in api result...
```

Comment:
>/ [](start = 3, length = 1)

I don't know what this means. Usually if there is an issue, MSALs through a UiRequiredException, which app developers catch and then issue an interactive request. With U/P request, however, there is no UI. So....I don't know what this comment means, do you know @henrik-me?

Reply:
FYI, this comment was from the original developer, so I'm not sure what it meant.

Reply:
ping @henrik-me

Reply:
Please delete the comment. 

Context: MSAL libraries have a fairly poor understanding of Kerberos / IntegratedWindowsAuth flow which is actually happening here. When there will be a better understanding, we will update all libs. AFAIK there is no indication that UI is required from this API.

Reply:
Done.